### PR TITLE
begin scaffolding to close #26

### LINKED ISSUE #26: Make the unicode-segmentation dependency optional
This crate is being used by a number of proc macros that are exceedingly unlikely to ever see non-ascii input. It would be good if the unicode-segmentation dependency would thus be optional.

### PR DESCRIPTION
**Pull request by efx**
This introduces a naive way to only enable one way of iterating over the boundaries. This seem simple enough but am quite open to better designs!. Next week I can try to implement the character splitting behavior

### CODE REVIEW

**Comment by jplatte** [comment]
`unicode` seems like an appropriate feature name. Have you seen the CI failure?

File: src/lib.rs [hunk 1]
```diff
@@ -62,8 +62,18 @@ pub use upper_camel::{
 
 use std::fmt;
 
+#[cfg(feature = "unicode")]
 use unicode_segmentation::UnicodeSegmentation;
```
Comment (jplatte) [inline src/lib.rs:66]: You can just move this import into `get_iterator`, right?

**Review by jplatte — APPROVED** [review]
Sorry this took so long. Looks good, just have one small suggestion.

File: src/lib.rs (hunk 1 above)
Comment (efx) [reply src/lib.rs:66]: Yes! great point.

**Pull request merged**